### DISCUSSION TITLE
Custom BusinessRule Needs to Access DataPortal

### DISCUSSION THREAD

**Discussion by MarkOverstreet** [MarimerLLC/csla#3260] [Questions] [answered]
So there are instances where we create a business rule that inherits from the BusinessRule class and in certain cases will need to go back to the server using the DataPortal.  In previous versions this was not an issue because we just used the static DataPortal class like ...

                            if (document.IsNew)
                            {
                                //Check if using hire date otherwise it is normal store only document
                                if (doesUseHireDate)
                                {
                                    GetOrganizationSettingCommand cmd = await GetOrganizationSettingCommand.NewAsync();
                                    cmd.OrganizationKey = orgKey;
                                    cmd.Name = "IgnoreHireDateOnUpload";
                                    cmd = DataPortal.Execute(cmd);

                                    //Check if organization has ignore hire date enabled
                                    if (!bool.Parse(cmd.Value)) //If we are checking for hire date
                                    {

But in the new version of CSLA you are using dependency injection so I have two questions:

1)  Should we be using the DataPortal within a BusinessRule?
2)  If yes, how can we inject it?  Do you have any examples??

Thanks for any assistance.

**Comment by saulyt** [comment, marked as answer]
IRuleContext has access to the ApplicationContext. You can therefore do something like this:
```
protected override async Task ExecuteAsync(Csla.Rules.IRuleContext context) 
{
	var portal = context.ApplicationContext.GetRequiredService<IDataPortal<GetOrganizationSettingCommand>>();
	var cmd = portal.Create();
	cmd.OrganizationKey = orgKey;
	cmd.Name = "IgnoreHireDateOnUpload";
	cmd = await portal.ExecuteAsync(cmd);
	//Check if organization has ignore hire date enabled........
}
```

**Reply by MarkOverstreet** [reply]
interesting....let me try that.   Thanks!
